store/setup.ts

```diff
@@ -2,7 +2,28 @@
 import { createSlice } from '@reduxjs/toolkit';
 import { lowerCase, orderBy } from 'lodash';
 
-const initialState = {
+type RankedValue = {
+  points: number;
+  rank: number;
+};
+
+type SetupState = {
+  account: string | null;
+  balance: number | null;
+  chainId: string | null;
+  errorMessage: string | null;
+  isVerified: boolean;
+  isLeaderboardLoading: boolean;
+  leaderboard: RankedValue[];
+  nftDetails: any | null;
+  isMemoryDetailsLoading: boolean;
+  memoryDetails: any[];
```

Comment:
we try to avoid using "any" as it disables type safety. can you try adding types for this? if don't have time, feel free to add TODOs